### PR TITLE
RET-3438: Fix for NOC email is not being sent to Removed Organisation

### PR DESCRIPTION
**Pull request by robin-ieong**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocRespondentRepresentativeService.java [hunk 1]
```diff
@@ -135,7 +143,21 @@ public void updateRepresentativesAccess(CallbackRequest callbackRequest) throws
             log.info("About to apply representation change {}", changeRequest);
 
             nocCcdService.updateCaseRepresentation(accessToken, changeRequest,
-                caseDetails.getJurisdiction(), caseDetails.getCaseTypeId(), caseDetails.getCaseId());
+                    caseDetails.getJurisdiction(), caseDetails.getCaseTypeId(), caseDetails.getCaseId());
+
+            String email = caseDataBefore.getRepCollection()
+                    .stream().filter(r -> r.getValue().getRespondentOrganisation().getOrganisationID()
+                            .equals(changeRequest.getOrganisationToRemove().getOrganisationID()))
```
Comment (paul-peasegood) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocRespondentRepresentativeService.java:150]: Is the spacing correct on the .equals line

could we put changeRequest.getOrganisationToRemove().getOrganisationID() in a variable?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationServiceTest.java [hunk 2]
```diff
@@ -113,8 +113,6 @@ void sendNotificationsShouldSendThreeNotifications() {
         nocNotificationService.sendNotificationOfChangeEmails(callbackRequest, caseData);
         // Claimant
         verify(emailService, times(1)).sendEmail(any(), eq("claimant@represented.com"), any());
-        // Previous respondent
-        verify(emailService, times(1)).sendEmail(any(), eq("oldres@rep.com"), any());
```
Comment (paul-peasegood) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationServiceTest.java:117]: Why remove this line? Could we set it to times(0) if no longer required

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocRespondentRepresentativeService.java (hunk 1 above)
Comment (robin-ieong) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocRespondentRepresentativeService.java]: Spacing looks to be correct, and yes refactored to use new variable

**Review by paul-peasegood — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationServiceTest.java (hunk 2 above)
Comment (robin-ieong) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationServiceTest.java:117]: changed to use times(0) as discussed

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationService.java [hunk 3]
```diff
@@ -46,17 +46,6 @@ public void sendNotificationOfChangeEmails(CallbackRequest callbackRequest, Case
...
-            log.warn("missing oldSolicitorEmail");
-        } else {
-            emailService.sendEmail(
-                previousRespondentSolicitorTemplateId,
-                oldSolicitorEmail,
-                NocNotificationHelper.buildPreviousRespondentSolicitorPersonalisation(caseData)
-            );
-        }
-
         String newSolicitorEmail = NocNotificationHelper.getNewSolicitorEmail(callbackRequest);
```
Comment (fergusoshea) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationService.java:49]: In this Helper, I think we may need to update the getNewSolicitorEmail() to use the new CaseData to find the index of the Solicitor?
54: getIndexOfSolicitor(oldCaseData) -> getIndexOfSolicitor(newDetails)

Comment (robin-ieong) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/NocNotificationService.java]: Refactored everything, thanks.

**Review by furnivall — APPROVED** [review]

**Pull request merged**